app/controllers/reporting_controller.rb

```diff
@@ -0,0 +1,56 @@
+class ReportingController < ActionController::API
+  before_action :build_recruitment_cycle
+  before_action :build_courses
+
+  def reporting
+    status = :ok
+
+    render status: status, json: course_stats
+  end
+
+private
+
+  def course_stats
+    {
+      total: {
+        all: @courses.count,
+        non_findable: @courses.count - @findable_courses.count,
+        all_findable: @findable_courses.count,
+      },
+      findable_total: {
+        open: @open_courses.count,
+        closed: @closed_courses.count,
+      },
+      provider_type: { **group_by_count(:provider_type) },
+      program_type: { **group_by_count(:program_type) },
+
+      study_mode: { **group_by_count(:study_mode) },
+      qualification: { **group_by_count(:qualification) },
+      is_send: { **group_by_count(:is_send) },
+    }
+  end
```

Comment:
I think this should go into a service or reporting class. Then you can break down the tests to one per stat, and keep them out of the request / feature specs.